mirascope/core/base/_call_factory.py

```diff
@@ -87,6 +87,7 @@ def call_factory(  # noqa: ANN202
     handle_stream_async: HandleStreamAsync[
         _AsyncResponseChunkT, _BaseCallResponseChunkT, _BaseToolT
     ],
+    provider: str | None = None,
```

Comment:
In order to hand over the provider name to Response, we decided to pass it 
from call_factory with the argument provider.
We can create new class from ClassResponse.
```python
class OpenAIBaseCallResponse(CallResponse):.
    _provider: str = "openai"
```
but I don't want to create new one for providers, the policy was to pass the name to call_factory.